tools/rewrite.ts

```diff
@@ -33,27 +37,60 @@ async function main() {
     return
   }
 
-  console.log('Rewriting LMDB cache to another LMDB cache');
+  LoggerFactory.INST.logLevel('error');
 
   const input = new LmdbCache<any>({ ...defaultCacheOptions, dbLocation: args.input })
     .storage<RootDatabase<any, string>>();
+
   const output = new LmdbCache<any>({ ...defaultCacheOptions, dbLocation: args.output })
     .storage<RootDatabase<any, string>>();
 
-  const bar = new cliProgress.SingleBar({
-    etaBuffer: 1000,
-  }, cliProgress.Presets.shades_classic);
-  bar.start(input.getKeysCount(), 0);
+  const numKeys = input.getKeysCount()
+  const chunkSize = 1000
+
+  const endChunk = Math.min(Math.ceil(numKeys / chunkSize), args.startChunk + args.numChunks)
+
+  let counter = 0
+  let i
+  for (i = args.startChunk; i < endChunk; i++) {
+    try {
+      output.transactionSync(() => {
+        // input.getKeys({ snapshot: false, limit: chunkSize, offset: i * chunkSize })
+        //   .forEach((key) => {
+        //     console.log(key)
+        //     let value = input.getBinary(key)
+        //     fs.writeFileSync("./problem.bin", value)
+        //     console.log(value)
+        //   });
+        input.getRange({ snapshot: false, limit: chunkSize, offset: i * chunkSize })
+          .forEach(({ key, value }) => {
+            try {
+              output.putSync(key, value)
+            } catch (err) {
+              console.error("Failed to put", key, "exception:", err)
+            }
+            counter += 1
+          });
+      })
+
+      await output.flushed
+
+    } catch (err) {
+      console.error("Failed to chunk, exception:", i, err)
+      process.exit(1)
+    }
+  }
+  i -= 1
+  console.log("Inserted elements ", i * chunkSize + counter, "of", numKeys, "(", Math.floor((100 * i * chunkSize + counter) / numKeys), "%)")
+  if (counter !== 0) {
```

Comment:
I don't understand how it works :-( the counter is only increased, but never decreased - so it always will be != 0?